lib/oppen.rb

```diff
@@ -22,4 +22,91 @@ def self.print(tokens: [], margin: 80, new_line: "\n")
     end
     printer.output
   end
+
+  # Wadler.
+  class Wadler
+    attr_reader :tokens
+    attr_reader :margin
+    attr_reader :new_line
+    attr_reader :current_nest_indentation
+
+    def initialize(margin: 80, new_line: "\n")
+      @tokens = []
+      @margin = margin
+      @new_line = new_line
+      @current_nest_indentation = 0
+    end
+
+    # <Description>
+    # @return [<Type>] <description>
+    def output
+      tokens.append(::Oppen::Token::EOF.new)
+      ::Oppen.print(tokens:, margin:, new_line:)
+    end
+
+    # <Description>
+    # @param indent [<Type>] <description>
+    # @param open_obj [<Type>] <description>
+    # @param close_obj [<Type>] <description>
+    # @param break_type [<Type>] <description>
+    #
+    # @return [<Type>] <description>
+    def group(indent: 0, open_obj: '', close_obj: '',
+              break_type: ::Oppen::Token::BreakType::CONSISTENT)
+      if open_obj != ''
+        self.break
+        text(open_obj)
+      end
+
+      nest(indent:, break_type:) { # rubocop:disable Style/ExplicitBlockArgument
+        yield
+      }
```

Comment:
I think it's better if we accept a block in `group` as an arg and pass it to nest, instead of calling `yield` here.

The linter is correct, this is _"bad style"_.